hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenFeeScheduleUpdateTransactionHandler.java

```diff
@@ -69,42 +71,61 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
      * Handles custom fees. Returns the list of collectors automatically associated with the newly created token if the
      * custom fees are from a token create transaction
      *
-     * @param customFeeList protobuf custom fee list
+     * @param protoCustomFees protobuf custom fee list
      * @return A list of collectors automatically associated with the token if it's a token create transaction
      */
     Set<EntityId> updateCustomFees(
-            Collection<com.hederahashgraph.api.proto.java.CustomFee> customFeeList,
+            Collection<com.hederahashgraph.api.proto.java.CustomFee> protoCustomFees,
             RecordItem recordItem,
             Transaction transaction) {
         var autoAssociatedAccounts = new HashSet<EntityId>();
         var consensusTimestamp = transaction.getConsensusTimestamp();
         var tokenId = transaction.getEntityId();
-        var id = new CustomFee.Id(consensusTimestamp, tokenId);
+        var customFee = new CustomFee();
+        customFee.setTokenId(tokenId.getId());
+        customFee.setTimestampRange(Range.atLeast(consensusTimestamp));
 
-        for (var protoCustomFee : customFeeList) {
-            var collector = EntityId.of(protoCustomFee.getFeeCollectorAccountId());
-            var customFee = new CustomFee();
-            customFee.setId(id);
-            customFee.setCollectorAccountId(collector);
-            customFee.setAllCollectorsAreExempt(protoCustomFee.getAllCollectorsAreExempt());
+        // For empty custom fees, add a single row with only the timestamp and tokenId.
+        if (protoCustomFees.isEmpty()) {
+            entityListener.onCustomFee(customFee);
+            return autoAssociatedAccounts;
+        }
 
+        for (var protoCustomFee : protoCustomFees) {
+            var allCollectorsAreExempt = protoCustomFee.getAllCollectorsAreExempt();
+            var collector = EntityId.of(protoCustomFee.getFeeCollectorAccountId());
             var feeCase = protoCustomFee.getFeeCase();
             boolean chargedInAttachedToken;
 
             switch (feeCase) {
                 case FIXED_FEE:
-                    chargedInAttachedToken = parseFixedFee(customFee, protoCustomFee.getFixedFee(), tokenId);
+                    var fixedFee = parseFixedFee(protoCustomFee.getFixedFee(), tokenId);
+                    fixedFee.setAllCollectorsAreExempt(allCollectorsAreExempt);
+                    fixedFee.setCollectorAccountId(collector);
+                    customFee.addFixedFee(fixedFee);
+                    recordItem.addEntityId(fixedFee.getDenominatingTokenId());
+                    chargedInAttachedToken = fixedFee.isChargedInToken(tokenId);
```

Comment:
add a `AbstractFee fee;` before the switch,

then change this to

```suggestion
                    var fixedFee = parseFixedFee(protoCustomFee.getFixedFee(), tokenId);
                    customFee.addFixedFee(fixedFee);
                    fee = fixedFee;
```

after the `switch` block,

do

```java
  fee.setAllCollectorsAreExempt(allCollectorsAreExempt);
  fee.setCollectorAccountId(collector);
  recordItem.addEntityid(fee.getDenominatingTokenId());
  boolean chargedInAttachedToken = fee.isChargedInToken(tokenId);
```

You'll need to add `getDenominatingTokenId()` to the abstract class, with default returning `EntityId.EMPTY`. For `RoyaltyFee`, the function returns fallbackFee.denominatingTokenId or `EntityId.EMPTY` if there's no fallback fee. You'll have to annotate it properly in the three concrete classes, so only in `FixedFee` the `denominating_token_id` appears in the top level json string.

Reply:
Sounds good I have made this update, there was a slight change to the example as denominatingTokenId is not in Abstract Fee.